### ISSUE TITLE
An item with the same key has already been added.

### ISSUE THREAD

**Issue by PeterTranMGM** [Azure/azure-functions-openapi-extension#407]
**Describe the issue**
This error comes up including an class in the same assembly.

**To Reproduce**
1. Clone https://github.com/PeterTranMGM/FunctionAppOpenApiSameKey
2. Run debugger
3. Access http://localhost:7071/api/swagger.json

**Expected behavior**
An actual JSON

**Screenshots**
If applicable, add screenshots to help explain your issue.

**Environment (please complete the following information, if applicable):**
 - OS: Windows
 - Package Version: 1.3.0

**Additional context**
Doesn't seem to happen in 1.2.0
Function App project is using out of process model

**Comment by william-liebenberg** [comment]
@justinyoo - I am getting the same issue right now :(

**Comment by mvacha** [comment]
We are having the exact same issue and also we managed to work around it by downgrading to previous version (.net 6, OOP fx app).

**Comment by yarus** [comment]
The same issue with .net 6, isolated azure function app).

```
    [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json",
        bodyType: typeof(UpdatedApplicationExternalResponse),
        Description = "")]
```

UpdatedApplicationExternalResponse type used only in 1 function definition.

**Comment by ihordyrman** [comment]
Have the same problem here. I found that problems have started from this point. For some reason, I have two similar models in the list and it tries to put all of them in one dictionary as a key
![image](https://user-images.githubusercontent.com/26085482/161744114-516ddab9-e23f-4ee6-a5ee-805452457468.png)

**Comment by ihordyrman** [comment]
It looks like the main problem is in DocumentHelperExtensions.GetHttpTriggerMethods. I have only 3 methods in my app, but it returns 2 identical objects for the same method.
![image](https://user-images.githubusercontent.com/26085482/161756634-6b2807cd-a1e3-4d26-bb11-c5322aa84768.png)

**Comment by brennfoster** [comment]
Getting the same error when using the same request body model across different functions. Even though the different functions are different types of requests (post, put). Downgraded to 1.2.0 to fix.

**Comment by svengrav** [comment]
> Getting the same error when using the same request body model across different functions. Even though the different functions are different types of requests (post, put). Downgraded to 1.2.0 to fix.

That is exactly my problem too. I use the same model for the post/update. This does not seem to work. But a low version doesn't help either.

**Comment by olehgerus** [comment]
I have the same issue using generic class as a wrapper for my responses. Downgrading didn't help me too.

**Comment by situssoft** [comment]
I had the same error a couple of days ago.

For my use case this error stems from the use of Assembly.LoadFrom/LoadFile in a couple of places 

- Microsoft.Azure.WebJobs.Extensions.OpenApi.Core\Extensions\AssemblyExtensions.cs line 69: GetTypesFromReferencedFunctionApps
- Microsoft.Azure.Functions.Worker.Extensions.OpenApi\OpenApiHttpTriggerContext.cs line 327: GetAssembly

The documentation for LoadFrom/LoadFile seems out of date with respect to how AssemblyLoadContext works, all the docs I am seeing appear to relate to the .NET Framework style of assembly loading. AssemblyLoadContext is explained [here](https://docs.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext).

The code at the first ref above has a TODO comment which kind of suggests we undertand there is more to do.

By changing the functions ref'd above to use AssemblyLoadContext.Default.LoadFromAssmblyPath / LoadFromAssemblyName i have worked around the error for my case.

However, there is a LOT of assembly loading going on in this code base none of which seems to take notice of the .NET Core AssemblyLoadContext (which could be a legacy problem) so I am far from confident that the above represents a suitable fix.

**Comment by tushargw** [comment]
> We are having the exact same issue and also we managed to work around it by downgrading to previous version (.net 6, OOP fx app).

For the records, I also implemented a bare bones .Net 6 isolated function in visual studio 2022, which returns a json (serialized object). I added swagger support and v1.3.0 failed for me. v1.2.0 worked for me. And at this time I didn't find any prerelease either beyond v1.3.0. Looking forward to that fix #426 making its way to prerelease.

**Comment by mshafiqmk** [comment]
> Getting the same error when using the same request body model across different functions. Even though the different functions are different types of requests (post, put). Downgraded to 1.2.0 to fix.

This works for me also.

Thanks [svengrav](https://github.com/svengrav)
